### PR TITLE
Encrypted Key Store (RPC API & CLI)

### PR DESCRIPTION
**Pull request by cryptoquick**
**Summary of changes**
Changes introduced in this pull request:
- [x] HTTP JSON-RPC API method

**Reference issue to close (if applicable)**

Closes: #1033 

**Other information and links**

Blocks: #1078

### CODE REVIEW

File: key_management/src/keystore.rs [hunk 1]
```diff
@@ -120,6 +120,13 @@ pub mod json {
     }
 }
 
+enum EncKsError {
```
Comment (ec2) [inline key_management/src/keystore.rs:123]: Whats EncKs?

Comment (ec2) [reply key_management/src/keystore.rs:123]: Oh duh, nvm. Encrypted Keystore

File: node/rpc/src/lib.rs [hunk 2]
```diff
@@ -232,6 +233,8 @@ where
                 "Filecoin.NetAddrsListen",
                 net_api::net_addrs_listen::<DB, KS, B>,
             )
+            // Encryption
+            .with_method("Filecoin.EncUnlock", enc_api::enc_unlock::<DB, KS, B>)
```
Comment (ec2) [inline node/rpc/src/lib.rs:237]: If these methods arent part of FIlecoin but Forest, i would rather call prefix them with `Forest.`, so we can better differentiate what's specifically ours and whats not.

**Review by connormullett — APPROVED** [review]
Thanks for the assist

**Comment by cryptoquick** [comment]
@connormullett Did you want to take care of the CLI portion?

**Comment by cryptoquick** [comment]
> > @connormullett Did you want to take care of the CLI portion?
> 
> The CLI portion as in the CLI command that calls the RPC methods? I can try but I might need some help.

Haha, no worries, I'll add that piece to this draft before I mark it as RFR. Feel free to lift anything to your branch if you need it.

**Comment by cryptoquick** [comment]
So, @connormullett's changes [here](https://github.com/ChainSafe/forest/pull/1078/commits/08ab041935c9e59486b5637b243b4a096834e617) I think are sufficient to satisfy the CLI portion of the feature, which won't be included here. Gonna mark this RFR, to merge into connor's branch.

**Comment by connormullett** [comment]
Going to go ahead and merge, I can address any issues on my branch.

**Pull request merged**